contracts/test/integration/IntegrationTestMaplePCVDeposit.t.sol

```diff
@@ -0,0 +1,643 @@
+        vm.expectEmit(true, false, false, false, address(usdcDeposit));
+        emit Harvest(MainnetAddresses.MPL_TOKEN, 12345, block.timestamp);
+        // check Harvest event for USDC interests (no interests/losses)
+        vm.expectEmit(true, false, false, true, address(usdcDeposit));
+        emit Harvest(MainnetAddresses.USDC, 0, block.timestamp);
+        // check Withdrawal event for principal
+        vm.expectEmit(true, true, false, true, address(usdcDeposit));
+        emit Withdrawal(
+            MainnetAddresses.GOVERNOR,
+            address(this),
+            targetUsdcBalance
+        );
+        // do withdraw
+        usdcDeposit.withdraw(address(this), targetUsdcBalance);
+
+        assertEq(usdcDeposit.balance(), 0);
+        assertEq(usdcDeposit.lastRecordedBalance(), 0);
+        assertEq(usdc.balanceOf(address(this)), targetUsdcBalance);
+        assertTrue(mapleToken.balanceOf(address(usdcDeposit)) > 0);
+    }
+
+    function testHarvest() public {
+        uint256 rewardRate = IMapleRewards(mapleRewards).rewardRate();
+        vm.prank(mapleOwner);
+        IMapleRewards(mapleRewards).notifyRewardAmount(rewardRate);
+
+        vm.warp(block.timestamp + IMaplePool(maplePool).lockupPeriod());
+
+        // check Harvest event for MPL rewards (don't check amount)
+        vm.expectEmit(true, false, false, false, address(usdcDeposit));
+        emit Harvest(MainnetAddresses.MPL_TOKEN, 12345, block.timestamp);
+        usdcDeposit.harvest();
+
+        uint256 mplBalance = mapleToken.balanceOf(address(usdcDeposit));
+
+        assertTrue(mplBalance != 0);
+    }
+
+    function testAccrueInterestsEarned() public {
+        assertEq(usdcDeposit.lastRecordedBalance(), targetUsdcBalance);
+
+        // Someone creates a loan
+        address mapleLoanFactory = 0x36a7350309B2Eb30F3B908aB0154851B5ED81db0;
+        address mapleDebtLockerFactory = 0xA83404CAA79989FfF1d84bA883a1b8187397866C;
+        address mapleLoan;
+        uint256 gracePeriod = 432000; // 5 days
+        uint256 paymentInterval = 15552000; // 180 days
+        bytes memory arguments = abi.encode(
+            address(this), // borrower
+            address(0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599), // assets[0] = collateral = WBTC
+            MainnetAddresses.USDC, // assets[1] = principal = USDC
+            gracePeriod, // terms[0]
+            paymentInterval, // terms[1] = paymentInterval
+            2, // terms[2] = numberOfPayments
+            0, // amounts[0] = collateralRequired
+            targetUsdcBalance, // amounts[1] = principalRequested
+            targetUsdcBalance, // amounts[2] = endingPrincipal
+            0.12e18, // rates[0] = interestRate = 12%
+            0.02e18, // rates[1] = earlyFeeRate = 2%
+            0, // rates[2] = lateFeeRate = 0
+            0.05e18 // rates[3] = lateInterestPremium = 5%
+        );
+        mapleLoan = IMapleLoanFactory(mapleLoanFactory).createInstance(
```

Comment:
I'm surprised that a mapleLoan can be created without pranking as a user address. I didn't know the platform was this permissionless.

Reply:
yep! it's up to the pool manager to fund it though, and they probably need legal paperwork before executing anything onchain